libs/testserver/fake_workspace.go

```diff
@@ -0,0 +1,237 @@
+package testserver
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+)
+
+// FakeWorkspace holds a state of a workspace for acceptance tests.
+type FakeWorkspace struct {
+	directories map[string]bool
+	files       map[string][]byte
+	// normally, ids are not sequential, but we make them sequential for deterministic diff
+	nextJobId int64
+	jobs      map[int64]jobs.Job
+}
+
+func NewFakeWorkspace() *FakeWorkspace {
+	return &FakeWorkspace{
+		directories: map[string]bool{
+			"/Workspace": true,
+		},
+		files:     map[string][]byte{},
+		jobs:      map[int64]jobs.Job{},
+		nextJobId: 1,
+	}
+}
+
+func (s *FakeWorkspace) ScimMe() (iam.User, int) {
+	return iam.User{
+		Id:       "1000012345",
+		UserName: "tester@databricks.com",
+	}, http.StatusOK
+}
+
+func (s *FakeWorkspace) CurrentMetastoreAssignment() (catalog.MetastoreAssignment, int) {
+	return catalog.MetastoreAssignment{
+		DefaultCatalogName: "main",
+	}, http.StatusOK
+}
+
+func (s *FakeWorkspace) DirectoryPermissions(objectId string) (workspace.WorkspaceObjectPermissions, int) {
+	return workspace.WorkspaceObjectPermissions{
+		ObjectId:   objectId,
+		ObjectType: "DIRECTORY",
+		AccessControlList: []workspace.WorkspaceObjectAccessControlResponse{
+			{
+				UserName: "tester@databricks.com",
+				AllPermissions: []workspace.WorkspaceObjectPermission{
+					{
+						PermissionLevel: "CAN_MANAGE",
+					},
+				},
+			},
+		},
+	}, http.StatusOK
+}
+
+func (s *FakeWorkspace) WorkspaceGetStatus(path string) (workspace.ObjectInfo, int) {
+	if s.directories[path] {
+		return workspace.ObjectInfo{
+			ObjectType: "DIRECTORY",
+			Path:       path,
+		}, http.StatusOK
+	} else if _, ok := s.files[path]; ok {
+		return workspace.ObjectInfo{
+			ObjectType: "FILE",
+			Path:       path,
+			Language:   "SCALA",
+		}, http.StatusOK
+	} else {
+		return workspace.ObjectInfo{}, http.StatusNotFound
+	}
+}
```

Comment:
I think we could keep all this new implementations in fake_workspace.go -- this is potentially generally useful.

however, hard-coded responses should stay in server_test.go. We'll probably move them to test.toml

Reply:
I thought that the right thing for hard-coded responses would be implementing CRUD. For instance, there is an API to create cluster policies and instead of using test.toml we can use CLI to create policies.